### PR TITLE
[LMS-12075] Bug fix for Date Deserializer

### PR DESCRIPTION
**Pull request by jedrekdomanski**
### PR desription

Fix typoo causing Date deserializer to error out.
```
Params deserialization error for mp.onboarding topic: undefined method `isa?' for Wed, 01 Dec 2021:Date
Did you mean?  is_a?
[[simple_payments_app] {}:] Exception raised when processing mp.onboarding/0 in offset range 0..0 -- Karafka::Errors::DeserializationError: undefined method `isa?' for Wed, 01 Dec 2021:Date
Did you mean?  is_a?
/home/jedrek/.rvm/gems/ruby-2.7.4/bundler/gems/field_struct_avro_schema-cb2e1e55dabd/lib/field_struct/avro_schema/karafka/deserializer.rb:26:in `rescue in call'
```

### Expected results
Deserialization is successful

```ruby
Params deserialization for mp.onboarding topic successful in 10 ms
{:guid=>"onbo-d21d327e-303d-4af6-96d7-5f21b59d5097", :lease=>{:guid=>"leas-d21d327e-303d-4af6-96d7-5f21b59d5097", :lease_number=>9162618, :total_of_renewal_payments=>2249.389892578125, :uncapped_total_of_renewal_payments=>2249.389892578125, :cash_amount=>999.9500122070312, :amount_approved=>2350.0, :initial_payment_amount=>50.0, :initial_payment_tax_request_id=>"txrq-9233f1ff-e0e7-4aff-a579-66f42d53552e", :initial_payment_amount_tax=>3.5, :ninety_day_buyout_amount=>1049.949951171875, :ninety_day_buyout_fee=>0.0, :pre_rent_buyout_days=>90, :pre_rent_buyout_fee=>0.0, :pre_rent_buyout_percentage=>0.0, :pre_rent_buyout_amount=>1049.949951171875, :pre_rent_buyout_type=>"flat_fee", :renewal_payment_amount=>43.2599983215332, :renewal_payment_frequency=>"weekly", :agreement_signed_through=>"simple_signature", :funded_date=>Wed, 01 Dec 2021, :delivery_date=>Wed, 01 Dec 2021, :notes=>[{:guid=>"note-c06b7f5c-a19a-43a8-b7bf-cdb52ad92fa5", :note=>"MCI (402) 502-5650, asked for 4 digit code, (Customer does not actually have a cell phone number, Only a text app number) spoke with cust and verified info, not bundled, EDD 11/14, This 4 digit code I am providing you is equivalent to signing the lease agreement. Please ask the retailer to provide you the opportunity to review the lease agreement or print you off a copy prior to signing.  Once you have read your agreement you may provide the four digit code to the Retailer... Your code is 4558 #4DigitOverPhone", :user=>{:guid=>"user-00a362ab-aaab-4775-94f7-7c46b3ae4692", :first_name=>"Zachary", :last_name=>"Offen", :email=>"zachary.offen@acimacredit.com", :created_at=>Fri, 02 Jul 2021 22:37:27 +0000}, :created_at=>Sun, 14 Nov 2021 22:47:14 +0000, :updated_at=>Sun, 14 Nov 2021 22:47:14 +0000}, {:guid=>"note-84b2100f-3591-43b2-b326-f8523254fa99", :note=>"MCI (402) 502-5650 Needs to update phone number to <[BANK/CARD: xxxxxxxxxxxx0255]> because <Typo>. Searched system matches no other leases. Re ran app.  Updated phone number from <[BANK/CARD: xxxxxxxxxxxx0255]>. \n", :user=>{:guid=>"user-00a362ab-aaab-4775-94f7-7c46b3ae4692", :first_name=>"Zachary", :last_name=>"Offen", :email=>"zachary.offen@acimacredit.com", :created_at=>Fri, 02 Jul 2021 22:37:27 +0000}, :created_at=>Sun, 14 Nov 2021 22:41:03 +0000, :updated_at=>Sun, 14 Nov 2021 22:41:03 +0000}], :merchandise_description=>"7PC LIVINGROOM SET", :merchant_industry_code=>"weel_tire", :contract_state=>"NE", :money_factor=>1.190000057220459, :daily_lease_rate=>6.299698829650879, :tax_enabled=>true, :calculations_version=>"V2", :omit_initial_payment=>false, :lease_agreement=>{:original_resource_path=>"b5vf0etu856gc3wjp7ehracfhhqk", :resource_path=>"7gct5atowul2d9h8il9ahuyyraby", :bucket_name=>"merchant-portal-prod"}, :origin=>"merchant_portal", :business_entity=>"acima-credit", :rent_scheme_override=>"", :early_purchase_option_percentage=>45, :lease_term=>365, :markup=>50.0, :loss_damage_waiver=>{:enabled=>false}, :unbanked=>{:enabled=>false}, :api_partner_name=>"Name", :unaffiliated=>false}, :customer=>{:guid=>"cust-f801501c-871d-48ad-8e70-f6aa29178497", :deposit_type=>"direct_deposit", :last_payday=>Fri, 12 Nov 2021, :next_payday=>Fri, 19 Nov 2021, :payment_methods=>{:primary_method=>"bank_account", :card=>{:guid=>"card-4b42b090-1b2e-4913-9bb6-cf3f2c13f99c"}, :bank_account=>{:guid=>"bank-51f5e525-1038-46b2-875b-95e7ae5a0b69", :account_type=>"checking", :account_number=>"[BANK ACCOUNT: xxxxxxxxxxxx3470]", :routing_number=>"104913912", :bank_name=>"Pinnacle Bank"}}}, :merchant=>{:guid=>"merc-29ae0967-3a2c-4166-b9bf-701d69a927a6", :name=>"7 Day Furniture", :location=>{:guid=>"loca-e94c3bac-39ed-4f2f-863a-ec06dcdfb315", :name=>"7 Day Furniture - Omaha", :address=>{:street1=>"4911 S 72nd St", :city=>"Omaha", :zip=>"68127", :state=>"NE", :street2=>"", :label=>""}}}}
Inline processing of topic mp.onboarding with 1 messages took 12 ms
1 messages on mp.onboarding topic delegated to Consumers::Mp::NewOnboards
[[simple_payments_app] {}:] Marking mp.onboarding/0:1 as processed
[[simple_payments_app] {}:] Committing offsets with recommit: mp.onboarding/0:2
```

### CODE REVIEW

**Review by bfreese — APPROVED** [review]
Looks good.  don't forget to bump up the version number and publish :)

**Comment by aemadrid** [comment]
@jedrekdomanski please do not merge PRs without all the reviewers approval. Even though this is a small PR and it is good work it is not good procedure to merge a PR without all approvers.

**Pull request merged**
